ooni/ui/web/server.py

```diff
@@ -0,0 +1,608 @@
+        except:
+            self.director_event_poller.notify(DirectorEvent("error",
+                                                            "Failed to start deck"))
+
+    @app.route('/api/nettest/<string:test_name>/start', methods=["POST"])
+    @xsrf_protect(check=True)
+    @requires_true(attrs=['_director_started', '_is_initialized'])
+    def api_nettest_start(self, request, test_name):
+        try:
+            _ = self.director.netTests[test_name]
+        except KeyError:
+            raise WebUIError(500, 'Could not find the specified test')
+
+        try:
+            test_options = json.load(request.content)
+        except ValueError:
+            raise WebUIError(500, 'Invalid JSON message recevied')
+
+        test_options["test_name"] = test_name
+        deck_data = {
+            "tasks": [
+                {"ooni": test_options}
+            ]
+        }
+        try:
+            deck = NGDeck()
+            deck.load(deck_data)
+            self.run_deck(deck)
+
+        except errors.MissingRequiredOption as option_name:
+            raise WebUIError(
+                400, 'Missing required option: "{}"'.format(option_name)
+            )
+
+        except usage.UsageError:
+            raise WebUIError(
+                400, 'Error in parsing options'
+            )
+
+        except errors.InsufficientPrivileges:
+            raise WebUIError(
+                400, 'Insufficient priviledges'
+            )
+        except:
+            raise WebUIError(
+                500, 'Failed to start nettest'
+            )
+
+        return self.render_json({"status": "started"}, request)
+
+    @app.route('/api/nettest', methods=["GET"])
+    @xsrf_protect(check=False)
+    @requires_true(attrs=['_director_started', '_is_initialized'])
+    def api_nettest_list(self, request):
+        return self.render_json(self.director.netTests, request)
+
+    @app.route('/api/input', methods=["GET"])
+    @xsrf_protect(check=False)
+    @requires_true(attrs=['_is_initialized'])
+    def api_input_list(self, request):
+        input_store_list = self.director.input_store.list()
+        for key, value in input_store_list.items():
+            value.pop('filepath')
+        return self.render_json(input_store_list, request)
+
+    @app.route('/api/input/<string:input_id>/content', methods=["GET"])
+    @xsrf_protect(check=False)
+    @requires_true(attrs=['_is_initialized'])
+    def api_input_content(self, request, input_id):
+        content = self.director.input_store.getContent(input_id)
+        request.setHeader('Content-Type', 'text/plain')
+        request.setHeader('Content-Length', len(content))
+        return content
+
+    @app.route('/api/input/<string:input_id>', methods=["GET"])
+    @xsrf_protect(check=False)
+    @requires_true(attrs=['_is_initialized'])
+    def api_input_details(self, request, input_id):
+        return self.render_json(
+            self.director.input_store.get(input_id), request
```

Comment:
Here I am not sure whether returning this input also returns the filepath, which above you try to hide